app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -239,38 +177,47 @@ def validate(self, data):
         return data
 
     def update(self, experiment, data):
-        primary_probe_set_ids = data.pop("primary_probe_set_ids", [])
-        secondary_probe_set_ids = data.pop("secondary_probe_set_ids", [])
-        experiment = super().update(experiment, data)
-
-        if primary_probe_set_ids or secondary_probe_set_ids:
-            with transaction.atomic():
+        with transaction.atomic():
+            if set(data.keys()).intersection(
+                {"primary_probe_set_ids", "secondary_probe_set_ids"}
+            ):
                 experiment.probe_sets.clear()
+
+            primary_probe_set_ids = data.pop("primary_probe_set_ids", [])
+            secondary_probe_set_ids = data.pop("secondary_probe_set_ids", [])
+            experiment = super().update(experiment, data)
+
+            if primary_probe_set_ids:
```

Comment:
Does this all need to be in a transaction? Looks like we have 2 transactions that start & end for every update, even if branches & probe sets aren't included in the update data